src/index.js

```diff
@@ -522,35 +538,7 @@ class TerserPlugin {
         });
       }
 
-      if (javascript && javascript.JavascriptModulesPlugin) {
-        const hooks = javascript.JavascriptModulesPlugin.getCompilationHooks(
-          compilation
-        );
-        const data = serialize({
-          terser: terserPackageJson.version,
-          terserOptions: this.options.terserOptions,
-        });
-
-        hooks.chunkHash.tap(plugin, (chunk, hash) => {
-          hash.update('TerserPlugin');
-          hash.update(data);
-        });
-      } else {
-        // Todo remove after drop `webpack@4` compatibility
-        const { mainTemplate, chunkTemplate } = compilation;
-        const data = serialize({
-          terser: terserPackageJson.version,
-          terserOptions: this.options.terserOptions,
-        });
-
-        // Regenerate `contenthash` for minified assets
-        for (const template of [mainTemplate, chunkTemplate]) {
-          template.hooks.hashForChunk.tap(plugin, (hash) => {
-            hash.update('TerserPlugin');
-            hash.update(data);
-          });
-        }
-      }
+      this.webpackCompat.updateAssetsHash(compilation, this.options);
```

Comment:
no need extra layer, let's keep this as is

Reply:
do you mean `if/else`?

Reply:
yes, keep as is